holoviews/core/spaces.py

```diff
@@ -686,11 +734,14 @@ def clone(self, data=None, shared_data=True, new_type=None, *args, **overrides):
 
         # Ensure the clone references this object to ensure
         # stream sources are inherited
-        if clone.callback is self.callback:
-            clone.callback = self.callback.clone()
-        if self not in clone.callback.inputs:
+        if isinstance(clone.callback, OverlayCallable):
+            from ..util import Dynamic
+            return Dynamic(clone, shared_data=shared_data)
+        elif clone.callback is self.callback:
+            clone.callback = clone.callback.clone()
+        if not any(inp is self for inputs in get_stream_sources(clone).values() for inp in inputs):
```

Comment:
How about ``if all(inp is not self ...``?

Why do you only do this if self isn't in the inputs?